lib/improver/spotdata_new/neighbour_finding.py

```diff
@@ -0,0 +1,517 @@
+        """
+        method_name = '{}{}{}'.format('nearest',
+                                      '_land' if self.land_constraint else '',
+                                      '_minimum_dz' if self.minimum_dz else '')
+        return method_name
+
+    def _transform_sites_coordinate_system(self, x_points, y_points, cube):
+        """
+        Function to convert coordinate pairs that specify spot sites into the
+        coordinate system of the model from which data will be extracted. Note
+        that the cartopy functionality returns a z-coordinate which we do not
+        want in this case, as such only the first two columns are returned.
+
+        Args:
+            x_points (np.array):
+                An array of x coordinates to be transformed in conjunction
+                with the corresponding y coordinates.
+            y_points (np.array):
+                An array of y coordinates to be transformed in conjunction
+                with the corresponding x coordinates.
+            cube (iris.cube.Cube):
+                A cube from the model from which data will be extracted. This
+                provides the coordinate system onto which the spot site's
+                coordinates should be remapped.
+        Returns:
+            np.array:
+                An array containing the x and y coordinates of the spot sites
+                in the target coordinate system, shaped as (n_sites, 2). The
+                z coordinate column is excluded from the return.
+        """
+        target_coordinate_system = cube.coord_system().as_cartopy_crs()
+
+        return target_coordinate_system.transform_points(
+            self.site_coordinate_system, x_points, y_points)[:, 0:2]
+
+    @staticmethod
+    def check_sites_are_within_domain(sites, site_coords, site_x_coords,
+                                      site_y_coords, cube):
+        """
+        A function to remove sites from consideration if they fall outside the
+        domain of the provided model cube. A warning is raised and the details
+        of each rejected site are printed.
+
+        Args:
+            sites (list of dicts):
+                A list of dictionaries defining the spot sites for which
+                neighbours are to be found.
+            site_coords (np.array):
+                An array of shape (n_sites, 2) that contains the spot site
+                coordinates in the coordinate system of the model cube.
+            site_x_coords (np.array):
+                The x coordinates of the spot sites in their original
+                coordinate system, from which invalid sites must be removed.
+            site_y_coords (np.array):
+                The y coordinates of the spot sites in their original
+                coordinate system, from which invalid sites must be removed.
+            cube (iris.cube.Cube):
+                A cube that is representative of the model/grid from which spot
+                data will be extracted.
```

Comment:
It doesn't modify in place (although I thought it would at first). I'm not sure why, because I am redefining the names internally (although pointing at a subset of themselves)?

Anyhow, I had to return things. Returns section added.